public/New-DbaDbUser.ps1

```diff
@@ -69,165 +74,246 @@ function New-DbaDbUser {
         [string[]]$Database,
         [string[]]$ExcludeDatabase,
-        [switch]$IncludeSystem = $False,
+        [switch]$IncludeSystem,
+        ### The following parameters are specific to the objects that the command works with, in this case a database user.
+        ### We start with the name of the object, in this case the name of the user that will be created.
+        ### Discussion: The SMO class is called "User", the SQL syntax to create the object uses "CREATE USER", the command is named "...User" - so the parameter should be renamed to User (with an alias for backwords compatility)
+        ### For the default parameter set Login, the name of the user can be set to the mandatory parameter Login, in all other cases, we need the name of the user.
+        [Parameter(ParameterSetName = "Login")]
+        [Parameter(Mandatory, ParameterSetName = "NoLogin")]
+        [Parameter(Mandatory, ParameterSetName = "ContainedSQLUser")]
+        [Parameter(Mandatory, ParameterSetName = "ContainedAADUser")]
+        [Alias("Username")]
+        [string]$User,
+        ### Now we add parameters to specify the individual attributes for the object. We start with parameters that are mandatory for parameter sets.
         [Parameter(Mandatory, ParameterSetName = "Login")]
         [string]$Login,
-        [Parameter(ParameterSetName = "Login", Mandatory = $False)]
-        [Parameter(Mandatory, ParameterSetName = "NoLogin")]
-        [Parameter(ParameterSetName = "ContainedSQLUser", Mandatory = $True)]
-        [Parameter(ParameterSetName = "ContainedAADUser", Mandatory = $True)]
-        [string]$Username,
-        [string]$DefaultSchema = 'dbo',
-        [Parameter(ParameterSetName = "ContainedSQLUser", Mandatory = $True)]
-        [securestring]$Password,
-        [Parameter(ParameterSetName = "ContainedAADUser", Mandatory = $True)]
+        ### If we need to pass a password to the command, we always use the type securestring and name the parameter SecurePassword. Here we only use the alias for backwords compatility.
+        [Parameter(Mandatory, ParameterSetName = "ContainedSQLUser")]
+        [Alias("Password")]
+        [securestring]$SecurePassword,
+        [Parameter(Mandatory, ParameterSetName = "ContainedAADUser")]
         [switch]$ExternalProvider,
+        ### Now we add parameters to specify the individual attributes for the object that are not specific for a parameter set.
+        ### As we want to use the schema dbo in most cases, we use default values in those cases.
+        [string]$DefaultSchema = 'dbo',
+        ### All commands that create new objects have a switch parameter Force to drop and re-create the object in case it already exists.
+        ### Sometimes, this parameter also changes the ConfirmPreference to "none". This way, -WhatIf and -Force cannot be used together. So this has to be removed everywhere.
+        ### This parameter is always the second last parameter.
```

Comment:
I think it's "second to last" instead of "second last"

Reply:
DeepL and PONS disagree with you...